### PR TITLE
Lm king (clean legal move-king branch)

### PR DESCRIPTION
**Pull request by MelanieJae**
Branch cleanup as discussed Monday, commented out all non-king models and tests; when merging, if conflict, remove commented out version

### CODE REVIEW

**Comment by MelanieJae** [comment]
Aaron: I kept the 'return true' s in the king logic b/c even when I changed assert to refute, they still failed. I used the general logic pattern we discussed on Monday but I think it still needs at least the 'return true' for reasons I still haven't figured out. Melanie

File: test/models/king_test.rb [hunk 1]
```diff
@@ -2,44 +2,28 @@
...
+    assert king.legal_move?(3, 5)
+    assert king.legal_move?(5, 5)
   end
 
-  test "king legal_move? 1 square diagonal downward left returns true" do
-    king = FactoryGirl.create(:king, :row_position => 4, :col_position => 7)
+  test "king legal_move? 1 square diagonal downward left or upward right returns true" do
+    king = FactoryGirl.create(:king, :row_position => 4, :col_position => 5)
     assert king.legal_move?(3, 6)
+    assert king.legal_move?(5, 5)
```
Comment (washburnad) [inline test/models/king_test.rb]: This is not a diagonal move, FYI.  5, 4 perhaps?

**Comment by washburnad** [comment]
Hi Melanie.  This looks good.  I'm not sure I understand your comment about removing return true.

I did make one comment above about refactoring, but it can be done later.

File: app/models/king.rb [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
+    return true if  (self.row_position - x) == -1 and (self.col_position - y) == 0
+      # move one step vertically
+    return true if  (self.row_position - x) == 0 and (self.col_position - y) == 1
+    return true if  (self.row_position - x) == 0 and (self.col_position - y) == -1
       # move one step diagonally
-      return true if (self.row_position - x).abs == 1 and (self.col_position - y).abs == 1
-    end
-    return false
+    return true if  (self.row_position - x).abs == 1 and (self.col_position - y).abs == 1
   end
```
Comment (washburnad) [inline app/models/king.rb]: what's true about all of these is the that the row_diff.abs <= 1 and col_diff.abs <= 1

**Comment by washburnad** [comment]
:+1:

**Pull request merged**